src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -1213,6 +1211,192 @@ private static float MinMaxCore<TMinMaxOperator>(ReadOnlySpan<float> x)
+
+                }
+
+                // Aggregate the lanes in the vector to create the final scalar result.
+                return TIndexOfMinMax.Invoke(max, resultIndex);
+            }
+
+            if (Vector128.IsHardwareAccelerated && x.Length >= Vector128<float>.Count)
+            {
+                ref float xRef = ref MemoryMarshal.GetReference(x);
+                Vector128<int> resultIndex = Vector128.Create(0, 1, 2, 3);
+                Vector128<int> curIndex = Vector128.Create(0, 1, 2, 3);
+                Vector128<int> increment = Vector128.Create(Vector128<float>.Count);
+
+                // Load the first vector as the initial set of results, and bail immediately
+                // to scalar handling if it contains any NaNs (which don't compare equally to themselves).
+                Vector128<float> result = Vector128.LoadUnsafe(ref xRef, 0), current;
+                if (!Vector128.EqualsAll(result, result))
+                {
+                    return GetFirstNaNIndex(result, resultIndex);
+                }
+
+                int oneVectorFromEnd = x.Length - Vector128<float>.Count;
+                int i = Vector128<float>.Count;
+
+                // Aggregate additional vectors into the result as long as there's at least one full vector left to process.
+                while (i <= oneVectorFromEnd)
+                {
+                    // Load the next vector, and early exit on NaN.
+                    current = Vector128.LoadUnsafe(ref xRef, (uint)i);
+                    curIndex = Vector128.Add(curIndex, increment);
+
+                    if (!Vector128.EqualsAll(current, current))
+                    {
+                        return GetFirstNaNIndex(current, curIndex);
+                    }
+
+                    TIndexOfMinMax.Invoke(ref result, current, ref resultIndex, curIndex);
+
+                    i += Vector128<float>.Count;
+                }
+
+                // If any elements remain, handle them in one final vector.
+                if (i != x.Length)
+                {
+                    curIndex = Vector128.Add(curIndex, Vector128.Create(x.Length - i));
+
+                    current = Vector128.LoadUnsafe(ref xRef, (uint)(x.Length - Vector128<float>.Count));
+                    if (!Vector128.EqualsAll(current, current))
+                    {
+                        return GetFirstNaNIndex(current, curIndex);
+                    }
+
+                    Vector128<float> remainderMask = CreateRemainderMaskSingleVector128(x.Length - i);
+
+                    TIndexOfMinMax.Invoke(ref result, current, ref resultIndex, curIndex, remainderMask);
+
+                }
+
+                // Aggregate the lanes in the vector to create the final scalar result.
+                return TIndexOfMinMax.Invoke(result, resultIndex);
+            }
+
+            // Scalar path used when either vectorization is not supported or the input is too small to vectorize.
+            {
```

Comment:
nit: Were braces here intentional?

Reply:
They were not. Leftovers that should have been cleaned up.